packages/devops/scripts/lambda/utilities.py

```diff
@@ -0,0 +1,236 @@
+                'Value': tupaia_instance_name
+            },
+        ],
+        Type=config['Type'],
+        IpAddressType=config['IpAddressType']
+    )
+    return response['LoadBalancers'][0]
+
+
+def create_gateway_target_group(tupaia_instance_name, config, target_group_name):
+    response = elbv2.create_target_group(
+        Name=target_group_name,
+        Protocol=config['Protocol'],
+        ProtocolVersion=config['ProtocolVersion'],
+        Port=config['Port'],
+        VpcId=config['VpcId'],
+        HealthCheckProtocol=config['HealthCheckProtocol'],
+        HealthCheckPort=config['HealthCheckPort'],
+        HealthCheckEnabled=config['HealthCheckEnabled'],
+        HealthCheckPath=config['HealthCheckPath'],
+        HealthCheckIntervalSeconds=config['HealthCheckIntervalSeconds'],
+        HealthCheckTimeoutSeconds=config['HealthCheckTimeoutSeconds'],
+        HealthyThresholdCount=config['HealthyThresholdCount'],
+        UnhealthyThresholdCount=config['UnhealthyThresholdCount'],
+        Matcher=config['Matcher'],
+        TargetType=config['TargetType'],
+        Tags=[
+            {
+                'Key': 'Type',
+                'Value': 'Tupaia-Gateway'
+            },
+            {
+                'Key': 'TupaiaInstanceName',
+                'Value': tupaia_instance_name
+            },
+        ]
+    )
+    return response['TargetGroups'][0]['TargetGroupArn']
+
+
+def create_gateway_listeners(elb_arn, target_group_arn, certificate_arn):
+    # HTTP
+    elbv2.create_listener(
+        LoadBalancerArn=elb_arn,
+        Protocol='HTTP',
+        Port=80,
+        DefaultActions=[
+            {
+                'Type': 'forward',
+                'ForwardConfig': {
+                    'TargetGroups': [
+                        {
+                            'TargetGroupArn': target_group_arn
+                        }
+                    ]
+                }
+            },
+        ]
+    )
+    # HTTPS
+    elbv2.create_listener(
+        LoadBalancerArn=elb_arn,
+        Protocol='HTTPS',
+        Port=443,
+        SslPolicy='ELBSecurityPolicy-2016-08',  # default
+        Certificates=[
+            {
+                'CertificateArn': certificate_arn
+            },
+        ],
+        DefaultActions=[
+            {
+                'Type': 'forward',
+                'ForwardConfig': {
+                    'TargetGroups': [
+                        {
+                            'TargetGroupArn': target_group_arn
+                        }
+                    ]
+                }
+            },
+        ]
+    )
+
+
+def register_gateway_target(target_group_arn, tupaia_instance_id):
+    elbv2.register_targets(
+        TargetGroupArn=target_group_arn,
+        Targets=[
+            {
+                'Id': tupaia_instance_id,
+                'Port': 80
+            },
+        ]
+    )
+
+
+def build_record_set_change(domain, subdomain, stage, gateway, ip_address):
```

Comment:
Could be worth either putting in big comment blocks separating the utilities that relate to various services, or even splitting them into different utility files. (general, route53, elb). Comment blocks is probably enough for now

Reply:
Added